src/lib/factories.tsx

```diff
@@ -195,7 +195,7 @@ function createShorthandFromValue(
   if (valIsReactElement) return React.cloneElement(value as React.ReactElement<Props>, props)
 
   // Create ReactElements from built up props
-  if (valIsPrimitive || valIsPropsObject) return <Component {...props} />
+  if (valIsPrimitive || valIsPropsObject) return React.createElement(Component, props)
```

Comment:
Is it a suite strange change? YES!

----

https://github.com/Microsoft/TypeScript/issues/28768
https://github.com/Microsoft/TypeScript/issues/28631#issuecomment-443194203